docs/source/api_reference/data_format.rst

```diff
@@ -0,0 +1,273 @@
+.. _data_format:
+
+File Format Specifications
+==========================
+
+.. toctree::
+    :maxdepth: 1
+    :hidden: false
+
+    Overview
+    Introduction
+    Description
+    Instructions
+    Illustration
+
+Overview
+--------
+This document has the following content:
+
+- Introduction: What is a ``.ts`` file, why and when to use it.
+- Description: What are the individual components of a ``.ts`` file.
+- Instructions: How to create your own ``.ts`` file.
+- Illustrations: A running example to tie up above sections.
+
+Introduction
+------------
+This document formalizes string identifiers used in ``.ts`` file format.
+Encoded in ``utf-8``, ``.ts`` files stores time-series dataset and its corresponding metadata and can be
+opened via any basic editor like notepad for visual inspection.
+String identifiers refer to strings beginning with ``@`` in the file.
+
+``.ts`` files contains information blocks in the following order:
+
+1. All the comments to describe the dataset. (begins with ``#`` in the file)
+2. All the String Identifiers to give information about the metadata (begins with ``@`` in the file).
+3. Actual dataset.
```

Comment:
note enough detail. "actual dataset" is not a concrete term. Only use concrete words and formulations. The meaning of these words is unclear.

Reply:
Do you have another term? "A dataset block" sounds equally confusing to me and I can't think of any other way of putting it.

Reply:
Like other blocks, this will also require a small description. I can't think of anything since I find it pretty self-explanatory. Also, what do we write in the detailed subsection for 'Dataset block'? This document is only meant to describe the first two blocks; adding information on how the dataset is supposed to be present seems like an unnecessary addition to me.

Not having a subsection for the last block is also weirdly asymmetric.

Reply:
> since I find it pretty self-explanatory

It's not self-explanatory. There is no clear reference frame, sorry.

Try to make it clearer what the block contains.